hedera-node/src/main/java/com/hedera/services/txns/contract/operation/HederaSStoreOperation.java

```diff
@@ -0,0 +1,88 @@
+package com.hedera.services.txns.contract.operation;
+
+import com.hedera.services.store.contracts.HederaWorldState;
+import com.hedera.services.store.contracts.HederaWorldUpdater;
+import com.hedera.services.txns.contract.gascalculator.GasCalculatorHedera_0_18_0;
+import org.apache.tuweni.units.bigints.UInt256;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.Gas;
+import org.hyperledger.besu.evm.account.MutableAccount;
+import org.hyperledger.besu.evm.frame.ExceptionalHaltReason;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.FrontierGasCalculator;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.operation.AbstractOperation;
+import org.hyperledger.besu.evm.operation.Operation;
+
+import java.util.Optional;
+
+public class HederaSStoreOperation extends AbstractOperation {
+
+	protected static final Operation.OperationResult ILLEGAL_STATE_CHANGE =
+			new Operation.OperationResult(
+					Optional.empty(), Optional.of(ExceptionalHaltReason.ILLEGAL_STATE_CHANGE));
+
+	private final boolean checkSuperCost;
+
+	public HederaSStoreOperation(final GasCalculator gasCalculator) {
+		super(0x55, "SSTORE", 2, 0, false, 1, gasCalculator);
+		checkSuperCost = !(gasCalculator instanceof GasCalculatorHedera_0_18_0);
+	}
+
+	@Override
+	public Operation.OperationResult execute(final MessageFrame frame, final EVM evm) {
+		final UInt256 key = UInt256.fromBytes(frame.popStackItem());
+		final UInt256 value = UInt256.fromBytes(frame.popStackItem());
+
+		final MutableAccount account =
+				frame.getWorldUpdater().getAccount(frame.getRecipientAddress()).getMutable();
+		if (account == null) {
+			return ILLEGAL_STATE_CHANGE;
+		}
+
+		UInt256 currentValue = account.getStorageValue(key);
+		boolean currentZero = currentValue.isZero();
+		boolean newZero = value.isZero();
+		Gas gasCost = Gas.ZERO;
+		if (currentZero && !newZero) {
+			HederaWorldState.WorldStateAccount hederaAccount =
+					((HederaWorldUpdater) frame.getWorldUpdater()).getHederaAccount(frame.getRecipientAddress());
+			if (hederaAccount == null) {
+				hederaAccount = ((HederaWorldUpdater) frame.getWorldUpdater()).getHederaAccount(
+						frame.getSenderAddress());
+			}
+			long durationInSeconds = Math.max(0, hederaAccount.getExpiry() - frame.getBlockValues().getTimestamp());
```

Comment:
If there is a nested contract creation, it will throw a NPE, because the account is not found neither in `provisionalAccountUpdates`, nor in the ledger.
This applies for `HederaCreateOperation`.
Example contract:

```solidity
pragma solidity ^0.7.0;

contract NestedChildren {
    address public child;
    event ChildCreated(address _address);
    constructor() {
        child = address(new Child());
        emit ChildCreated(address(child));
    }

    function callCreate() public {
        child = address(new Child());
        emit ChildCreated(address(child));
    }
}

contract Child {
    address public child;
    event AnotherChild(address _address);
    constructor() {
        child = address(new ThirdChild());
        emit AnotherChild(address(child));
    }
}

contract ThirdChild {

}
```
We've thought about saving the expiries as context variables.

Reply:
I think some of these errors are captured in `NewOpInConstructorSuite` - which is failing on the swap branch

Reply:
Those are **one-level** depth contract creation, here it is a two-level depth contract creation.

Reply:
I will check them as well.